app/src/common/shared/com/igalia/wolvic/search/SearchEngineWrapper.kt

```diff
@@ -85,89 +96,111 @@ class SearchEngineWrapper private constructor(aContext: Context) :
         mContextRef = WeakReference(aContext)
         mPrefs = PreferenceManager.getDefaultSharedPreferences(context!!)
         mAutocompleteEnabled = SettingsStore.getInstance(context!!).isAutocompleteEnabled
-        val preferredSearchEngineId = SettingsStore.getInstance(context!!).searchEngineId
-        setupSearchEngine(aContext, preferredSearchEngineId)
+        initSearchEngine()
     }
 
     val availableSearchEngines: Collection<SearchEngine>
-        get() = mSearchEnginesMap!!.values
+        get() {
+            updateSearchEngine()
+            return mSearchEnginesMap!!.values
+        }
 
     fun setDefaultSearchEngine() {
         if (hasContext()) setupSearchEngine(context!!, null)
     }
 
     fun setCurrentSearchEngineId(context: Context?, searchEngineId: String?) {
-        if (hasContext()) SettingsStore.getInstance(this.context!!).searchEngineId = searchEngineId
+        if (hasContext()) SettingsStore.getInstance(context!!).searchEngineId = searchEngineId
+    }
+
+    fun initSearchEngine() {
```

Comment:
Bikeshedding a bit but I think this should be called `setupPreferredSearchEngine()` as it's more descriptive

Reply:
Changed